_posts/en/newsletters/2019-05-29-newsletter.md

```diff
@@ -0,0 +1,566 @@
+technique that contract protocol researchers call a *covenant*.
+The primary described benefit of this proposed opcode is allowing a
+small transaction to be confirmed now (when fees might be high) and have that transaction
+trustlessly guarantee that a set of people will receive their actual
+payments later when fees might be lower.  This can make it much more
+economical for organizations that already implement techniques such as
+[payment batching][] to handle sudden fee spikes.
+
+Before we look at the new opcode itself, let's take a moment to look at
+how you might accomplish something similar using current Bitcoin
+transaction features.
+
+Alice wants to pay a set of ten people but transaction fees are
+currently high so that she doesn't want to send ten separate
+transactions or even use payment batching to send one transaction
+that includes an output for each of the receivers.  Instead, she wants
+to trustlessly commit to paying them in the future without having to pay
+onchain fees for ten outputs now.  So Alice asks each of the receivers
+for one of their public keys and creates an unsigned and unbroadcast *setup* transaction
+that pays those keys using a 10-of-10 multisig script.  Then she creates
+a child transaction that spends from the multisig output to the
+10 outputs she originally wanted to create.  We'll call this child
+transaction the *distribution* transaction.  She asks all the receivers
+to sign this distribution transaction and she ensures each person
+receives everyone else's signatures, then she signs and broadcasts the
+setup transaction.
+
+![A setup transaction paying a pre-signed distribution transaction](/img/posts/2019-05-tx-tree-1.png)
+
+When the setup transaction receives a reasonable number of
+confirmations, there's no way for Alice to take back her payment to the
+10 receivers.  As long as each of the receivers has a copy of the
+distribution transaction and all the others' signatures, there's also no
+way for any receivers to cheat any other receiver out of a payment.  So
+even though the distribution transaction that actually pays the
+receivers hasn't been broadcast or confirmed, the payments are
+secured by the confirmed setup transaction.  At any time, any
+of the receivers who wants to spend their money can broadcast
+the distribution transaction and wait for it to confirm.
+
+This technique allows spenders and receivers to lock in a set of
+payments during high fees and then only distribute the actual payments when
+fees are lower.  According to Bitcoin Core fee estimates at the time of
+writing, anyone patient enough to wait a week for a transaction to
+confirm (like the distribution transaction above) can save significantly on fees.
+Let's look at the example above in that context.  To make later
+comparisons to Taproot more fair, we'll assume some form of key and
+signature aggregation is being used, such as [MuSig][] or (in theory)
+multiparty ECDSA (see [Newsletter #18][]).
```

Comment:
https://github.com/JeremyRubin/lazuli/blob/master/doc/protocol.pdf is my earlier citation on doing it with ECDSA and shows how messy it is